openedx/core/djangoapps/discussions/plugins.py

```diff
@@ -0,0 +1,55 @@
+"""
+Course app configuration for discussions.
+"""
+from typing import Dict, Optional
+
+from django.contrib.auth import get_user_model
+from django.utils.translation import ugettext_noop
+from opaque_keys.edx.keys import CourseKey
+
+from openedx.core.djangoapps.course_apps.plugin import CourseApp
+from openedx.core.djangoapps.discussions.models import DiscussionsConfiguration
+
+User = get_user_model()
+
+
+class DiscussionCourseApp(CourseApp):
+    """Course App config for Discussions."""
+
+    app_id = "discussion"
+    name = ugettext_noop("Discussion")
+    description = ugettext_noop("Encourage participation and engagement in your course with discussion forums.")
+    default_enabled = True
+
+    @classmethod
+    def is_available(cls, course_key: CourseKey) -> bool:
+        """Discussions is always available."""
+        return True
+
+    @classmethod
+    def is_enabled(cls, course_key: CourseKey) -> bool:
+        """Discussions enable/disable status is stored in a separate model."""
+        return DiscussionsConfiguration.is_enabled(course_key)
```

Comment:
This will return `False` even if discussions for course are not configured. Is this expected?

Reply:
Yes. I don't think we want to consider unconfigured discussions to be enabled.

Reply:
@xitij2000, got it. I asked only because I can use `Discussions` on devstack even with this returning `False`.

Reply:
This is for the new style discussions app. I guess this also supports the legacy discussions, so we should support that as well, but AFAIK you can now configure legacy discussions, so it should just work.